### PR TITLE
perf(logs): moving base64 decode to log receiver

### PR DESCRIPTION
**Pull request by davidvader**
preparing for future log updates, moving base64 decode to when log is received, rather than on view/render.
this should improve performance greatly and goes in hand with additional incoming enhancements

### CODE REVIEW

File: src/elm/Pages/Build/Logs.elm [hunk 1]
```diff
@@ -321,18 +320,6 @@ logFocusExists steps =
         /= ( Nothing, Nothing )
 
 
-{-| base64Decode : returns a string from a Maybe Log and decodes it from base64
```
Comment (davidvader) [inline src/elm/Pages/Build/Logs.elm:324]: moving this to Util.elm

File: src/elm/Pages/Build/View.elm [hunk 2]
```diff
@@ -305,14 +304,11 @@ viewLogs org repo buildNumber step logs follow shiftDown =
 viewLogLines : Org -> Repo -> BuildNumber -> StepNumber -> LogFocus -> Maybe (WebData Log) -> Int -> Bool -> Html Msg
 viewLogLines org repo buildNumber stepNumber logFocus maybeLog following shiftDown =
     let
-        log =
+        decodedLog =
```
Comment (davidvader) [inline src/elm/Pages/Build/View.elm:307]: log is now decoded in Main.elm, no need to decode it in view/render

File: src/elm/Main.elm [hunk 3]
```diff
@@ -763,8 +763,11 @@ update msg model =
 
                             else
                                 Cmd.none
+
+                        decodedLog =
```
Comment (davidvader) [inline src/elm/Main.elm:767]: this is the meat of the enhancement, decode the log and overwrite the data, no need to do this in view/render and we shouldnt

**Review by wass3r — APPROVED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**